File: tier1/java/b8438abf/Handler.java
import java.net.URI;
import java.util.Arrays;
import java.util.List;

public final class Handler {
    private static final List<String> ALLOWED = Arrays.asList("api.example.com");

    public static String handle(String target) {
        URI uri = URI.create(target);
        if (!"https".equals(uri.getScheme()) || uri.getHost() == null
            || !ALLOWED.contains(uri.getHost())) {
            return "";
        }
        return fetch(uri);
    }

    private static String fetch(URI uri) {
        return uri.toString();
    }
}

File: tier1/javascript/f6746cf4/handler.js
const ALLOWED = new Set(['api.example.com']);

function handle(target) {
  let parsed;
  try {
    parsed = new URL(target);
  } catch (e) {
    return '';
  }
  if (parsed.protocol !== 'https:' || !ALLOWED.has(parsed.hostname)) {
    return '';
  }
  return fetchIt(parsed.toString());
}

function fetchIt(target) {
  return String(target);
}

module.exports = { handle };
